src/extractor/extractor_callbacks.cpp

```diff
@@ -242,16 +242,15 @@ void ExtractorCallbacks::ProcessWay(const osmium::Way &input_way, const Extracti
         }
     };
 
-    // Deduplicates street names and street destination names based on the street_map map.
+    // Deduplicates street names, destination names and refs based on the string_map map.
     // In case we do not already store the name, inserts (name, id) tuple and return id.
     // Otherwise fetches the id based on the name and returns it without insertion.
     const auto turn_lane_id_forward = requestId(parsed_way.turn_lanes_forward);
     const auto turn_lane_id_backward = requestId(parsed_way.turn_lanes_backward);
 
     const constexpr auto MAX_STRING_LENGTH = 255u;
-    // Get the unique identifier for the street name
-    // Get the unique identifier for the street name and destination
-    const auto name_iterator = string_map.find(MapKey(parsed_way.name, parsed_way.destinations));
+    // Get the unique identifier for the street name, destination, and ref
+    const auto name_iterator = string_map.find(MapKey(parsed_way.name, parsed_way.destinations, parsed_way.ref));
```

Comment:
By now all the `GetNameForID`, `name_iterator`, `name_id`s are semantically wrong and probably should be called something like data id or metadata id.